github-connector/internal/registration/servicedetailsbuilder.go

```diff
@@ -0,0 +1,77 @@
+package registration
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/kyma-incubator/hack-showcase/github-connector/internal/apperrors"
+)
+
+const (
+	specificationURL          = "https://raw.githubusercontent.com/colunira/github-openapi/master/githubopenAPI.json"
+	applicationRegistryPrefix = "http://application-registry-external-api.kyma-integration.svc.cluster.local:8081/"
+	applicationRegistrySuffix = "-app/v1/metadata/services"
+	applicationName           = "GITHUB_CONNECTOR_NAME"
+)
+
+//OSCommunicator is an interface used to allow mocking file reading
+type OSCommunicator interface {
```

Comment:
In general I'm not sure if it is a good idea to create such an interface. It has two responsibilities:
- read the name of the API to register (available in GITHUB_CONNECTOR_NAME env var)
- read a file with json

Those are two very different things. 

What I propose is:
- API name passed in argument to NewServiceDetailsBuilder
- OSComunicator renamed to FileReader (or ConfigReader) with one method matching os.Reader interface (https://golang.org/pkg/os/) 

With such a design:
- getting name of the API is a detail handled in a code using ServiceDetailsBuilder
- value returned from os.Open("..") can be directly passed to NewServiceDetailsBuilder

Please se the following resource: https://matthias-endler.de/2018/go-io-testing/ to learn how to mock io.Reader interface